src/Neuron/Web/View.hs

```diff
@@ -55,11 +56,11 @@ searchScript = $(embedStringFile "./src-js/search.js")
 helloScript :: Text
 helloScript = $(embedStringFile "./src-purescript/hello/index.js")
 
-mkSearchURI :: MonadThrow m => Maybe Text -> [Text] -> m URI
+mkSearchURI :: MonadThrow m => Maybe Text -> [Tag] -> m URI
```

Comment:
What's this `Maybe Text`? If it is not used, it should be removed entirely.

Reply:
That was in the web search PR, but it avoids URLs like `search.html?q=&tag=science` which look ugly. Instead the `q` query parameter is not included at all in the generated URL.

Reply:
There is only one place where it is called, and with a value of `Nothing` (never a `Just` value). What would happen if you just remove this argument? (And I never saw any "q" parameter in neuron; where is it used actually?).

Reply:
Neuron compiles fine with this change: 

```diff
diff --git a/src/Neuron/Web/View.hs b/src/Neuron/Web/View.hs
index f602dfd..bd7cea3 100644
--- a/src/Neuron/Web/View.hs
+++ b/src/Neuron/Web/View.hs
@@ -55,13 +55,12 @@ searchScript = $(embedStringFile "./src-js/search.js")
 helloScript :: Text
 helloScript = $(embedStringFile "./src-purescript/hello/index.js")
 
-mkSearchURI :: MonadThrow m => Maybe Text -> [Text] -> m URI
-mkSearchURI terms tags = do
+mkSearchURI :: MonadThrow m => [Text] -> m URI
+mkSearchURI tags = do
   let mkParam k v = URI.QueryParam <$> URI.mkQueryKey k <*> URI.mkQueryValue v
-      qParams = maybeToList (fmap (mkParam "q") terms)
       tagParams = fmap (mkParam "tag") tags
   route <- URI.mkPathPiece (Rib.routeUrlRel Route_Search)
-  params <- sequenceA (qParams ++ tagParams)
+  params <- sequenceA tagParams
   pure
     emptyURI
       { uriPath = Just (False, [route]),
@@ -69,14 +68,14 @@ mkSearchURI terms tags = do
       }
 
 -- TODO: render error message when the query is invalid
-mkSearchQuery :: Maybe Text -> [Text] -> Text
-mkSearchQuery terms tags =
+mkSearchQuery :: [Text] -> Text
+mkSearchQuery tags =
   fromMaybe
     (Rib.routeUrlRel Route_Search)
-    (URI.render <$> mkSearchURI terms tags)
+    (URI.render <$> mkSearchURI tags)
 
 mkSingleTagQuery :: Text -> Text
-mkSingleTagQuery tag = mkSearchQuery Nothing [tag]
+mkSingleTagQuery tag = mkSearchQuery [tag]
 
 renderRouteHead :: Monad m => Config -> Route store graph a -> store -> HtmlT m ()
 renderRouteHead config r val = do
```

Reply:
I'll do this change later on `master`. Nevermind. Let's keep this PR focused.

Reply:
Alright, not really related to the PR but I'll do it 👍